src/main/scala/symsim/examples/concrete/cliffwalking/CliffWalking.scala

```diff
@@ -0,0 +1,115 @@
+package symsim
+package examples.concrete.cliffwalking
+
+import cats.{Eq, Foldable, Monad}
+import cats.kernel.BoundedEnumerable
+
+import org.scalacheck.{Arbitrary, Gen}
+
+import symsim.concrete.Randomized
+
+case class CWState (x: Int, y: Int):
+    override def toString: String = s"($x,$y)"
+
+type CWObservableState = CWState
+type CWReward = Double
+
+enum CWAction:
+    case Up, Down, Left, Right
+
+    override def toString: String = this match
+        case CWAction.Down => "↓"
+        case CWAction.Up => "↑"
+        case CWAction.Left => "←"
+        case CWAction.Right => "→"
+
+object CliffWalking
+  extends Agent[CWState, CWObservableState, CWAction, CWReward, Randomized]
+    with Episodic:
+
+    /** The episode should be guaranteed to terminate after
+      * TimeHorizon steps. This is used *only* *for* testing. It does
+      * not actually termintae the episodes. It is a bug if they run
+      * longer.
+      */
+    val TimeHorizon: Int = 2000
+
+    def isFinal (s: CWState): Boolean =
+        s.y == 0 && s.x > 0
+
+    def observe (s: CWState): CWObservableState =
+        CWState (s.x, s.y)
+
+    def move (s: CWState, a: CWAction): CWState =
+        require(s.x >= 0)
+        require(s.x <= 11)
+        require(s.y >= 0)
+        require(s.y <= 3)
+
+        a match
+            case CWAction.Up => CWState (s.x, (s.y + 1).min(3))
+            case CWAction.Down => CWState (s.x, (s.y - 1).max(0))
+            case CWAction.Right => CWState ((s.x + 1).min(11), s.y)
+            case CWAction.Left => CWState ((s.x - 1).max(0), s.y)
+
+
+    def CWReward (s: CWState) (a: CWAction): CWReward =
```

Comment:
this is a local function so its name should not be capitalized. Why just not `def reward`?

Reply:
I followed the other examples type, so I changed it to "def cwReward".